src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  function adder(num) {
+    if (num !== undefined) {
+      result += num;
+
+      return adder;
+    } else {
+      const finalResult = result;
```

Comment:
```suggestion
      return adder;
    }
    
    const finalResult = result;
```
`else` is redundant here as the code will not go further after return anyway